src/chat.rs

```diff
@@ -38,7 +39,11 @@ use crate::mimeparser::SystemMessage;
 use crate::param::{Param, Params};
 use crate::peerstate::{Peerstate, PeerstateVerifiedStatus};
 use crate::sql;
-use crate::stock::StockMessage;
+use crate::stock::{
```

Comment:
+1, initially i thought that `File::stock_str()` calls a method added to the File class or so.

also, i am wondering why not doing a class `StockStr` or so and have methods as `file()`, `savedMessages()` etc. that take the correct arguments and return the respective string. having a dedicated enum for every string feels weird to me, at least this is the first reaction, maybe that is wrong :)

Reply:
Yeah, there's a few choices that could be made.  I also considered prefixing the types with `StockStr`, but in the end I went with using `::stock_str()` as the method being close enough.  I actually like your suggestion though, especially because of the name conflicts.   Perhaps even rename the module to `stock_str`?  And then I'd also change the method itself to be shorter: `stock_str::E2eAvailable::msg(context)`.
Does that sound reasonable?

Reply:
wrt shorter methods: `StockStr::e2eAvailable(context)` would be even shorter. and also that would probably reduce the enum-boilerplate for every single string.

however, i would also be fine with `stock_str::E2eAvailable::msg(context)`, but i would leave the final decision up to ppl knowing more about rust-conventions than me, so up to @link2xt and @flub here :)